macro/profiler/src/lib.rs

```diff
@@ -0,0 +1,2 @@
+#[cfg(not(target_arch = "riscv32"))]
```

Comment:
You should add a nop version for `riscv32` so that you can compile either way.

Reply:
I'm not sure. Compilation failure is less confusing.

Reply:
I made module not linked to guest , so by default it won't be available at guest.

Reply:
For pprof, it works on Windows, Macos and Linux. So it works with popular `aarch64` and `amd64` platforms. Do you want me switch from deny `riscv32` to only allow `aarch64` and `amd64`? The former is shorter, the later is a bit longer.

Reply:
Yeah I would think it should be allowlist.

Reply:
added in https://github.com/nexus-xyz/nexus-zkvm/commit/35f9b6bcffeebf7684445d1c159ae64591576eb8